hwcpipe.h

```diff
@@ -30,6 +30,8 @@
 #include <functional>
 #include <memory>
 
+#define API_CALL __attribute__((visibility("default")))
```

Comment:
should use https://cmake.org/cmake/help/v3.0/module/GenerateExportHeader.html

Reply:
Hm... this appears to be quite tricky.
GenerateExportHeader will create export header in bin directory, so I'll need to do some code to copy it over (or include) so it can be included in a build step. Including bin dir sounds like a nasty hack and then this CMake file doesn't have install step to copy it over. I am not really passionate about writing cmake code to handle all these cases.
Anyway, HWCPipe is designed to be used with Linux operating systems and relies heavily on system APIs, so I don't see adding platform-specific macro API_CALL as a problem here

Reply:
Fair enough. I don't feel strong about this. I don't think the file needs to be copied around though. First you can provide a folder path where to generate this header. Second as long as you are using cmake build system to build hwcpipe, cmake will make sure it can see the header. So it doesn't really matter where it generates it.

I am fine with this as it is for now.

Reply:
`generate_export_headers` doesn't seem to have an option to tell CMake where to generate export file, it has only EXPORT_FILE_NAME, not EXPORT_FILE_PATH. I can give it a try later, though.
>Second as long as you are using cmake build system to build hwcpipe, cmake will make sure it can see the header. So it doesn't really matter where it generates it.
It unfortunately matters since it will work only when building the library itself, but wouldn't work when you build a project that depends on that library - you need to provide an include path to hwcpipe_export.h.